pkg/datastore/row.go

```diff
@@ -0,0 +1,154 @@
+package datastore
+
+import (
+	"sync"
+
+	"github.com/operator-framework/operator-marketplace/pkg/apis/marketplace/v1alpha1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func newOperatorSourceRowMap() *operatorSourceRowMap {
+	return &operatorSourceRowMap{
+		Sources: map[types.UID]*operatorSourceRow{},
+	}
+}
+
+// OperatorSourceKey is what datastore uses to relate to an OperatorSource
+// object.
+type OperatorSourceKey struct {
+	// UID is the UID associated with the OperatorSource object.
+	UID types.UID
```

Comment:
Why do we save this field in the struct? If we are creating a map that uses this value as a key, isn't it enough that the map itself contains the key? Why store it here and then never reference the field?

Reply:
Good point, I think my next PR uses it. If you want I can remove it now and then add it back in the next PR.

Reply:
Yeah, I think it would make more sense that way.

Reply:
I also don't understand why we need the `UID` and `NamespacedName`. Given that both are unique shouldn't only one of them be required?

Reply:
datastore uses the UID in its methods and we need the namespaced name to query the object via the client. datastore can be changed to use the Namespaced name only, but I would leave it to a future refactor. 
@kevinrizza removed the UID from this PR, I will add it back the next.